File: LibraryProject/video.cpp
#include"video.h"
#include<iostream>
#include<string>
#include<vector>
#include "boost/algorithm/string.hpp"


using namespace std;
using namespace boost;

Video::Video(const string& call_number,const string& title,const string& subjects,
             const string& description,const string& distributor,const string& notes,
             const string& series,const string& label):Media(call_number,title,subjects,notes),
           description(description),distributor(distributor),series(series),label(label)
{

}

void Video::display() const
{
    cout << "-----------VIDEO----------" << endl;
    cout << "CALL NUMBER::" << call_number << endl << "TITLE::" << title << endl << "SUBJECTS::"
     << subjects << endl << "DESCRIPTION::" << description << endl << "DISTRIBUTOR::" <<
     distributor << endl << "NOTES::" << notes << endl << "SERIES::" << series << endl
    << "LABEL::" << label << endl;
}

bool Video::compare_other(const string& searchby) const
{
    size_t found = description.find(searchby);
    size_t found1 = notes.find(searchby);
    size_t found2 = distributor.find(searchby);
    if((found!=string::npos) || (found1!=string::npos) || (found2!=string::npos))
    {
        return true;
    }
    else
    {
        return false;
    }
}

File: Python-Library-Project/video.py
#Sumeet Tulsani
from media import Media

class Video(Media):
    def __init__(self,call_number,title,subjects,description,distributor,notes,series,label):
        Media.__init__(self,call_number,title,subjects,notes)
        self.description = description
        self.distributor = distributor
        self.series = series
        self.label = label

    def display(self):
        print ("---------------VIDEO-----------------");
        print ("CALL NUMBER::",self.call_number);
        print ("TITLE::",self.title);
        print ("SUBJECTS::",self.subjects);
        print ("DESCRIPTION::",self.description);
        print ("DISTRIBUTOR::",self.distributor);
        print ("SERIES::",self.series);
        print ("LABEL::",self.label);

    def compare_other(self,searchby):
        if (self.description.find(searchby) != -1) or  (self.notes.find(searchby) != -1) or (self.distributor.find(searchby) != -1):
            return True
        else:
            return False
